### PR TITLE
Some cleanup and modernization in EgammaIsolationAlgos plugins

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

* follows up on https://github.com/cms-sw/cmssw/pull/32651
* get rid of `getByLabel` calls
* convert to `global` plugins
* merge `EleIsoDetIdCollectionProducer` and `GamIsoDetIdCollectionProducer` to templated `EgammaIsoDetIdCollectionProducer`
* remove unused `EgammaElectronTkNumIsolationProducer`
* implement `fillDescriptions` in `ParticleBasedIsoProducer`

#### PR validation:

CMSSW compiles, local matrix tests pass.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by guitargeek** [comment]
I forgot to update some config files. The problem didn't get caught by my local tests because the former `particleBasedIsoProducer_cfi` was still sticking around in my environment as `particleBasedIsoProducer_cfi.pyc`. With the added third commit, the bot tests should succeed.

**Comment by perrotta** [comment]
@cmsbuild please test

File: RecoEgamma/EgammaIsolationAlgos/plugins/EgammaTowerIsolationProducer.cc [hunk 1]
```diff
@@ -21,65 +21,52 @@
...
 #include "FWCore/Framework/interface/MakerMacros.h"
 DEFINE_FWK_MODULE(EgammaTowerIsolationProducer);
 
-EgammaTowerIsolationProducer::EgammaTowerIsolationProducer(const edm::ParameterSet& config) : conf_(config) {
-  // use configuration file to setup input/output collection names
-  emObjectProducer_ = conf_.getParameter<edm::InputTag>("emObjectProducer");
-
-  towerProducer_ = conf_.getParameter<edm::InputTag>("towerProducer");
+EgammaTowerIsolationProducer::EgammaTowerIsolationProducer(const edm::ParameterSet& config)
+    : emObjectProducer_{consumes(config.getParameter<edm::InputTag>("emObjectProducer"))}
```
Comment (jpata) [inline RecoEgamma/EgammaIsolationAlgos/plugins/EgammaTowerIsolationProducer.cc:44]: curious how clang-format left this extra whitespace here, which looks rather odd

File: RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc [hunk 2]
```diff
@@ -184,3 +188,27 @@ void ParticleBasedIsoProducer::produce(edm::Event& theEvent, const edm::EventSet
...
+  desc.add<edm::InputTag>("electronTmpProducer", edm::InputTag("gedGsfElectronsTmp"));
+  desc.add<edm::InputTag>("pfCandidates", edm::InputTag("particleFlow"));
+  desc.add<std::string>("valueMapElePFblockIso", "gedGsfElectrons");
+  desc.add<edm::InputTag>("electronProducer", edm::InputTag("gedGsfElectrons"));
+  desc.add<edm::InputTag>("photonTmpProducer", edm::InputTag("gedPhotonsTmp"));
+  desc.add<edm::InputTag>("pfEgammaCandidates", edm::InputTag("particleFlowEGamma"));
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("ComponentName", "pfBlockBasedIsolation");
+    psd0.add<double>("coneSize", 9999999999);
```
Comment (jpata) [inline RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc:206]: could something like `std::numeric_limits<double>::max()` be better here?

File: RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc [hunk 3]
```diff
@@ -184,3 +188,27 @@ void ParticleBasedIsoProducer::produce(edm::Event& theEvent, const edm::EventSet
...
+    desc.add<edm::ParameterSetDescription>("pfBlockBasedIsolationSetUp", psd0);
+  }
+  desc.add<edm::InputTag>("photonProducer", edm::InputTag("gedPhotons"));
+  desc.add<std::string>("valueMapPhoPFblockIso", "gedPhotons");
+  descriptions.add("particleBasedIsolation", desc);
```
Comment (jpata) [inline RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc:211]: just curious why you are renaming if from the default label `particleBasedIsoProducer -> particleBasedIsolation`?

**Review by jpata — COMMENTED** [review]
A few minor comments/questions. I don't see any issues in the tests.

File: RecoEgamma/EgammaIsolationAlgos/plugins/EgammaTowerIsolationProducer.cc (hunk 1 above)
Comment (guitargeek) [reply RecoEgamma/EgammaIsolationAlgos/plugins/EgammaTowerIsolationProducer.cc:44]: That happens, unfortunately. I'm fixing it manually.

File: RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc (hunk 2 above)
Comment (guitargeek) [reply RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc:206]: Good idea!

File: RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc (hunk 3 above)
Comment (guitargeek) [reply RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc:211]: With the automatically generated `cfi` files, the plugin label is also the filename. Before, the filename was `particleBasedIsoProducer_cfi.py` and the label was particleBasedIsolation. I chose to stick with the same label, if that's okay.

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by guitargeek** [comment]
Seems like the tests didn't start here, right?

**Comment by jpata** [comment]
@cmsbuild please test

File: RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc (hunk 2 above)
Comment (guitargeek) [reply RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc]: ...but it didn't work. I'll revert this change

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
+reconstruction
- technical
- no relevant changes in tests

**Comment by qliphy** [comment]
+1

**Pull request merged**